ipatests/test_webui/test_hbac.py

```diff
@@ -138,29 +138,30 @@ def test_hbac_test(self):
         self.wait_for_request(n=2)
         self.assert_facet('hbactest', 'run_test')
         self.button_click('run_test')
-        self.wait_for_request(n=2)
         self.assert_text("div.hbac-test-result-panel p", 'Access Granted'.upper())
         self.button_click('prev')
         self.assert_facet('hbactest', 'rules')
         self.switch_to_facet('run_test')
+        self.wait_for_request(n=2)
         self.button_click('new_test')
         self.assert_facet('hbactest', 'user')
 
         # test pre-run validation and navigation to related facet
-        self.switch_to_facet('run_test')
-        self.button_click('run_test')
-        self.assert_dialog('message_dialog')
+        def __hbac_run_test(self):
```

Comment:
I wonder if there can be a better name for the method. It switches to a tab and clicks on 'run_tests'. But it sounds like it was running some UI test.